src/bitflags_trait.rs

```diff
@@ -0,0 +1,68 @@
+use core as _core;
+
+pub trait BitFlags<T>:
+    _core::ops::BitOr<Output = Self>
```

Comment:
I think it might be better for the `BitFlags` trait not to include these additional supertraits and let consumers add them as additional bounds if they need them.

On the one hand it seems reasonable that if you've got flags you'll want to perform bitwise operations on them, but on the other it's a bit of a grab-bag snapshot of our current functionality that we won't be able to extend (since we can't add new supertraits without breakage).

It might be best to stick with just the current set of methods on the trait which cover:

- Construction from bits
- Conversion back into bits
- Set operations on pairs of flags

That should be enough to support a wide range of fundamental use-cases.

Reply:
What about Clone + Copy? I need those to implement the iterator which is based on a value and removes the flags as it advances through it.

Reply:
Do we need the `BitFlags` type itself to be `Copy`? Or just the underlying bits? If I understand right, we could just perform shifts and masks on the bits themselves and use `BitFlags::from_bits_unchecked` to return just that flag.